src/plugins/visualize/common/locator.ts

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import type { SerializableRecord, Serializable } from '@kbn/utility-types';
+import { omitBy } from 'lodash';
+import type { ParsedQuery } from 'query-string';
+import { stringify } from 'query-string';
+import rison from 'rison-node';
+import type { Filter, Query, RefreshInterval, TimeRange } from 'src/plugins/data/common';
+import type { LocatorDefinition, LocatorPublic } from 'src/plugins/share/common';
+import { isFilterPinned } from '../../data/common';
+import { url } from '../../kibana_utils/common';
+import { GLOBAL_STATE_STORAGE_KEY, STATE_STORAGE_KEY, VisualizeConstants } from './constants';
+import { PureVisState } from './types';
+
+const removeEmptyKeys = (o: Record<string, Serializable>): Record<string, Serializable> =>
+  omitBy(o, (v) => v == null);
+
+// eslint-disable-next-line @typescript-eslint/consistent-type-definitions
+export type VisualizeLocatorParams = {
+  /**
+   * The ID of the saved visualization to load.
+   */
+  visId?: string;
+
+  /**
+   * Type of visualization.
+   *
+   * @note This is required to navigate to "create" page (i.e., when no `visId` has been provided).
+   */
+  type?: string;
```

Comment:
It looks like `type` is also available in `PureVisState` under the `vis` field.  Not sure if the references that need this could use that field instead.

Reply:
Good point, let's remove it from the locator params 👌🏻